dd-sdk-android/src/main/kotlin/com/datadog/android/webview/internal/rum/WebViewRumEventConsumer.kt

```diff
@@ -63,11 +62,11 @@ internal class WebViewRumEventConsumer(
         return event
     }
 
-    private fun getOffset(viewId: String): Long {
+    private fun getOffset(viewId: String, datadogContext: DatadogContext): Long {
         var offset = offsets[viewId]
         if (offset == null) {
-            offset = timeProvider.getServerOffsetMillis()
-            offsets[viewId] = offset
+            offset = datadogContext.time.serverTimeOffsetMs
+            synchronized(offsets) { offsets[viewId] = offset }
```

Comment:
why do you use `synchronized` block here ?

Reply:
because `offsets` field can be accessed from the different threads (it gets called from EWC, which can be invoked on the different worker threads), so we need to make the work with this field thread-safe.

The work happens on the worker thread anyway, so synchronisation penalty is not important.

Reply:
Yeah, I had that one before but then looked under the hood and noticed that Android uses a SingleThreadExecutor for the JS bridge, I tested it also and made sure we only have one thread there.

Reply:
I see the reason now actually with the new architecture, good point.